brave/browser-laptop
# Change URL suggestions to consider last access timestamp and access count

* Re-order results to show history items before bookmark items
  * Record number of times site is visited
  * Re-order history items based on number of times visited and last access timestamp
  * Update and create tests

Fixes: #3049

Auditors: @bbondy

Test plan:

A new attribute, count, is added to history items in the sites section of save-session-1. This attribute records the number of times a site has been accessed. The test plan is broken in two components, one to verify the count attribute is incremented on access, the second to verify that the count and last access time are used when sorting history suggestions.

Scenario 1

Part A

  1. Clear save-session-1
  2. Access a site
  3. Close browser
  4. Verify that the count attribute for the saved site in save-session-1 has a value of 1

Part B

  1. Open browser
  2. Navigate to site from Part A
  3. Close browser
  4. Verify that the count attribute for the saved site in save-session-1 has an incremented value

Scenario 2

  1. Clear save-session-1
  2. Open browser
  3. Visit gm.com and gm.ca
  4. Close browser
  5. Verify that the count attributes of the two new sites are identical (1)
  6. Open browser to a new tab
  7. Enter ‘gm’ into the url bar
  8. Verify that the most recently visited of the two gm sites is the first entry in the history auto-suggest list
